app/models/manageiq/providers/redhat/infra_manager/refresh_parser.rb

```diff
@@ -393,6 +401,12 @@ def self.vm_inv_to_hashes(inv, _storage_inv, storage_uids, cluster_uids, host_ui
       host_id = vm_inv.attributes.fetch_path(:placement_policy, :host, :id) if host_id.blank?
       host = host_uids.values.detect { |h| h[:uid_ems] == host_id } unless host_id.blank?
 
+      # If the vm has a host but the refresh does not include it in the "hosts" hash
```

Comment:
I would elaborate here a bit.  Instead of just saying what the code here is doing, include a quick comment as to why.  Something about the migrated VMs and targeted refresh not getting host information separately.

I won't hold up the review on this ...